tests/php/BlockTypes/ProductQuery.php

```diff
@@ -0,0 +1,429 @@
+				'compare' => 'IN',
+				'key'     => '_stock_status',
+			),
+			$merged_query['meta_query']
+		);
+
+		$parsed_block = $this->get_base_parsed_block();
+		$parsed_block['attrs']['query']['__woocommerceStockStatus'] = array(
+			'instock',
+			'onbackorder',
+		);
+		$this->block_instance->set_parsed_block( $parsed_block );
+
+		$merged_query = $this->block_instance->build_query( $parsed_block['attrs']['query'] );
+
+		$this->assertContainsEquals(
+			array(
+				'compare' => 'IN',
+				'key'     => '_stock_status',
+				'value'   => array( 'instock', 'onbackorder' ),
+			),
+			$merged_query['meta_query']
+		);
+	}
+
+	/**
+	 * Test merging attribute queries.
+	 */
+	public function test_merging_attribute_queries() {
+		$parsed_block = $this->get_base_parsed_block();
+		$parsed_block['attrs']['query']['__woocommerceAttributes'] = array(
+			array(
+				'taxonomy' => 'pa_test',
+				'termId'   => 1,
+			),
+			array(
+				'taxonomy' => 'pa_test',
+				'termId'   => 2,
+			),
+			array(
+				'taxonomy' => 'pa_another_test',
+				'termId'   => 3,
+			),
+		);
+
+		$merged_query = $this->initialize_merged_query( $parsed_block );
+
+		$this->assertContainsEquals(
+			array(
+				'field'    => 'term_id',
+				'terms'    => array( 3 ),
+				'operator' => 'IN',
+				'taxonomy' => 'pa_another_test',
+			),
+			$merged_query['tax_query']
+		);
+
+		$this->assertContainsEquals(
+			array(
+				'taxonomy' => 'pa_test',
+				'field'    => 'term_id',
+				'terms'    => array( 1, 2 ),
+				'operator' => 'IN',
+			),
+			$merged_query['tax_query']
+		);
+	}
+
+	/**
+	 * Test merging order by rating queries.
+	 */
+	public function test_merging_order_by_rating_queries() {
+		$parsed_block                              = $this->get_base_parsed_block();
+		$parsed_block['attrs']['query']['orderBy'] = 'rating';
+
+		$merged_query = $this->initialize_merged_query( $parsed_block );
+
+		$this->assertEquals( 'meta_value_num', $merged_query['orderby'] );
+		$this->assertEquals( '_wc_average_rating', $merged_query['meta_key'] );
+	}
+
+	/**
+	 * Test merging order by popularity queries.
+	 */
+	public function test_merging_order_by_popularity_queries() {
+		$parsed_block                              = $this->get_base_parsed_block();
+		$parsed_block['attrs']['query']['orderBy'] = 'popularity';
+
+		$merged_query = $this->initialize_merged_query( $parsed_block );
+
+		$this->assertEquals( 'meta_value_num', $merged_query['orderby'] );
+		$this->assertEquals( 'total_sales', $merged_query['meta_key'] );
+	}
+
+	/**
+	 * Test merging multiple queries.
+	 */
+	public function test_merging_multiple_queries() {
+		$parsed_block                              = $this->get_base_parsed_block();
+		$parsed_block['attrs']['query']['orderBy'] = 'rating';
+		$parsed_block['attrs']['query']['__woocommerceStockStatus'] = array(
+			'instock',
+			'onbackorder',
+		);
+		$parsed_block['attrs']['query']['__woocommerceAttributes']  = array(
```

Comment:
I'm a bit confused by the `=` alignments here. Line 208 is aligned with 209, and 210 with 214. Why are they not all aligned? Does PHP Lint not complain here? These rules are confusing 😅

Reply:
> Does PHP Lint not complain here?

`phpcs` will complains if we align `=` signs, I agree that's they're confusing 😂 

<img width="848" alt="image" src="https://user-images.githubusercontent.com/5423135/206335903-ce13c6dd-cd9a-4783-9d45-6225fab77424.png">